File: app/Models/Pegawai.php
<?php
namespace App\Models;

use Illuminate\Database\Eloquent\Model;

class Pegawai extends Model
{
    protected $fillable = [
        'nama', 'jabatan', 'tanggal_masuk', 'upah_harian', 'upah_mingguan',
        'tunjangan_transport', 'persen_jamsostek', 'ikut_jamsostek',
    ];

    protected $casts = [
        'ikut_jamsostek' => 'boolean',
    ];

    public function sebagaiSupir()
    {
        return $this->hasMany(Pengiriman::class, 'supir_id');
    }

    public function sebagaiKernet()
    {
        return $this->hasMany(Pengiriman::class, 'kernet_id');
    }

    public function slipGajis()
    {
        return $this->hasMany(SlipGaji::class);
    }

    public function akun()
    {
        return $this->hasOne(User::class);
    }
}

File: Models/Pegawai.cs
using System.Text.Json.Serialization;
namespace SistemPengirimanApp.Models;
public class Pegawai
{
    [JsonPropertyName("id")]
    public int Id { get; set; }

    [JsonPropertyName("nama")]
    public string Nama { get; set; } = string.Empty;

    [JsonPropertyName("jabatan")]
    public string Jabatan { get; set; } = "sopir"; // "sopir" atau "kernet"
    [JsonPropertyName("tanggal_masuk")]
    public DateTime? TanggalMasuk { get; set; }
    [JsonPropertyName("upah_harian")]
    public decimal UpahHarian { get; set; }
    [JsonPropertyName("upah_mingguan")]
    public decimal UpahMingguan { get; set; }
    [JsonPropertyName("tunjangan_transport")]
    public decimal TunjanganTransport { get; set; }
    [JsonPropertyName("persen_jamsostek")]
    public decimal PersenJamsostek { get; set; } = 4;
    [JsonPropertyName("ikut_jamsostek")]
    public bool IkutJamsostek { get; set; } = true;

    public override string ToString() => Nama;
}